File: frontend/src/types/insight.ts
export interface InsightUploadSummary {
  id: number;
  filename: string;
  row_count: number;
  uploaded_at: string;
  skipped_rows?: number;
}

export interface InsightSummary {
  total_posts: number;
  total_views: number;
  total_viewers: number;
  total_interactions: number;
  total_comments: number;
  total_shares: number;
  total_saves: number;
  total_reactions: number;
  total_impressions: number;
  avg_retention_pct: number | null;
}

export interface InsightPost {
  post_id: string;
  title: string;
  post_type: string | null;
  permalink: string | null;
  posted_at: string | null;
  duration_sec: number | null;
  views: number;
  viewers: number;
  interactions: number;
  comments: number;
  shares: number;
  saves: number;
  reactions: number;
  retention_pct: number | null;
}

export interface PostTypeBreakdown {
  post_type: string;
  post_count: number;
  total_views: number;
  avg_views: number;
  avg_retention_pct: number | null;
}

export interface TrendPoint {
  upload_id: number;
  uploaded_at: string;
  filename: string;
  total_views: number;
  total_interactions: number;
}

File: backend/app/schemas/insight.py
from datetime import datetime

from pydantic import BaseModel


class InsightUploadOut(BaseModel):
    id: int
    filename: str
    row_count: int
    uploaded_at: datetime
    skipped_rows: int = 0


class InsightSummaryOut(BaseModel):
    total_posts: int
    total_views: int
    total_viewers: int
    total_interactions: int
    total_comments: int
    total_shares: int
    total_saves: int
    total_reactions: int
    total_impressions: int
    avg_retention_pct: float | None


class InsightPostOut(BaseModel):
    post_id: str
    title: str
    post_type: str | None
    permalink: str | None
    posted_at: datetime | None
    duration_sec: float | None
    views: int
    viewers: int
    interactions: int
    comments: int
    shares: int
    saves: int
    reactions: int
    retention_pct: float | None


class PostTypeBreakdownOut(BaseModel):
    post_type: str
    post_count: int
    total_views: int
    avg_views: float
    avg_retention_pct: float | None


class TrendPointOut(BaseModel):
    upload_id: int
    uploaded_at: datetime
    filename: str
    total_views: int
    total_interactions: int
